aliasmgr.go

```diff
@@ -0,0 +1,446 @@
+package lnd
+
+import (
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/htlcswitch/hop"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// aliasBucket stores aliases as keys and their base SCIDs as values.
+	// This is used to populate the maps that the aliasMgr uses. The keys
+	// are alias SCIDs and the values are their respective base SCIDs. This
+	// is used instead of the other way around (base -> alias...) because
+	// updating an alias would require fetching all the existing aliases,
+	// adding another one, and then flushing the write to disk. This is
+	// inefficient compared to N 1:1 mappings at the cost of marginally
+	// more disk space.
```

Comment:
You can't give out the same aliases to peers currently in spec. They can give _you_ the same alias but that doesn't matter since it's only in the peer mapping which is outpoint->alias and only used in the invoice generation/sending channel updates